Assignment 2-Yawei Zhu.Rmd

```diff
@@ -86,15 +88,50 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
 ## Part II
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
-
-
-
+```{r}
+t2<-EDGE%>%select(from,count)%>%group_by(from)%>%summarize(sum(count))
+names(t2)<-c("id","count")
+s2<-EDGE%>%select(to,count)%>%group_by(to)%>%summarize(sum(count))
+names(s2)<-c("id","count")
+INFO<-bind_rows(t2,s2)
+VERTEX<-full_join(VERTEX, INFO, by="id")
+g2<-graph.data.frame(EDGE, vertices=VERTEX, directed=TRUE)
+plot(g2, layout=layout.fruchterman.reingold, edge.arrow.size=0.3,edge.width=EDGE$count, vertex.color=VERTEX$gender, vertex.size=VERTEX$count, vertex.shapes=as.factor(VERTEX$group))
```

Comment:
Great graph! You may want to size the vertices using a `log()` function so that the difference between teachers and students is so extreme.